[CI] FieldExtractorIT testScaledFloat failing

**Build Scans:**
- [elasticsearch-intake #8992 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/2zgdxolfp5rfu)
- [elasticsearch-pull-request #28601 / 8.15.1_bwc-snapshots](https://gradle-enterprise.elastic.co/s/4zgqdegpv5ua4)
- [elasticsearch-pull-request #28588 / 8.15.1_bwc-snapshots](https://gradle-enterprise.elastic.co/s/pcxzp4daiek7e)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.16.0#javaRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.FieldExtractorIT" -Dtests.method="testScaledFloat" -Dtests.seed=F223888C61480165 -Dtests.bwc=true -Dtests.locale=en-AU -Dtests.timezone=Africa/Nairobi -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testScaledFloat'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.qa.mixed.FieldExtractorIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Expected a map containing
columns: a list containing
        0: a map containing
       type: "double"
       name: "scaled_float_field"
 values: a list containing
        0: a list containing
          0: expected a numeric value within <9.9E-324> of <0.0> but <NaN> differed by <NaN> more than delta <9.9E-324>
```

**Issue Reasons:**
- [main] 3 failures in test testScaledFloat (0.8% fail rate in 359 executions)
- [main] 2 failures in step 8.15.1_bwc-snapshots (2.2% fail rate in 91 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (2.6% fail rate in 76 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -101,9 +101,6 @@ tests:
 - class: org.elasticsearch.xpack.restart.CoreFullClusterRestartIT
   method: testSnapshotRestore {cluster=UPGRADED}
   issue: https://github.com/elastic/elasticsearch/issues/111799
-- class: org.elasticsearch.xpack.esql.qa.mixed.FieldExtractorIT
-  method: testScaledFloat
-  issue: https://github.com/elastic/elasticsearch/issues/112003
 - class: org.elasticsearch.xpack.inference.InferenceRestIT
   method: test {p0=inference/80_random_rerank_retriever/Random rerank retriever predictably shuffles results}
   issue: https://github.com/elastic/elasticsearch/issues/111999

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java
@@ -220,7 +220,8 @@ public abstract class FieldExtractorTestCase extends ESRestTestCase {
 
     public void testScaledFloat() throws IOException {
         double value = randomBoolean() ? randomDoubleBetween(-Double.MAX_VALUE, Double.MAX_VALUE, true) : randomFloat();
-        double scalingFactor = randomDoubleBetween(0, Double.MAX_VALUE, false);
+        // Scale factors less than about 5.6e-309 will result in NaN (due to 1/scaleFactor being infinity)
+        double scalingFactor = randomDoubleBetween(1e-308, Double.MAX_VALUE, false);
         new Test("scaled_float").expectedType("double")
             .randomIgnoreMalformedUnlessSynthetic()
             .randomDocValuesUnlessSynthetic()

